crates/sui-indexer-alt-graphql/src/api/types/transaction/mod.rs

```diff
@@ -158,6 +179,159 @@ impl Transaction {
             contents,
         }))
     }
+
+    /// Cursor based pagination through transactions based on filters.
+    pub(crate) async fn paginate(
+        ctx: &Context<'_>,
+        scope: Scope,
+        page: Page<CTransaction>,
+        filter: TransactionFilter,
+    ) -> Result<Connection<String, Transaction>, RpcError> {
+        let mut conn = Connection::new(false, false);
+
+        if page.limit() == 0 {
+            return Ok(Connection::new(false, false));
+        }
+
+        let watermarks: &Arc<Watermarks> = ctx.data()?;
+
+        let reader_lo = watermarks.pipeline_lo_watermark("tx_digests")?.checkpoint();
+
+        let global_tx_hi = watermarks.high_watermark().checkpoint();
```

Comment:
Shouldn't this be:

```suggestion
        let global_tx_hi = watermarks.high_watermark().transaction();
```

? 

Maybe we need some more test coverage where there are multiple transactions per checkpoint to cover these cases.

Reply:
Yes, my bad thanks for spotting that. I'll add some test to cover this case.